types/nullable_test.go

```diff
@@ -0,0 +1,248 @@
+		},
+
+		{
+			name:      "simple object: set name to null value",
+			jsonInput: []byte(`{"name":null}`),
+			wantNull:  true,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: do not provide name in json data",
+			jsonInput: []byte(`{}`),
+			wantNull:  false,
+			wantSet:   false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t1 *testing.T) {
+			var obj SimpleString
+			err := json.Unmarshal(tt.jsonInput, &obj)
+			assert.NoError(t, err)
+			assert.Equalf(t, tt.wantNull, obj.Name.IsNull(), "IsNull()")
+			assert.Equalf(t, tt.wantSet, obj.Name.IsSet(), "IsSet()")
+			fmt.Println(obj.Name.Get())
+		})
+	}
+}
+
+type SimpleInt struct {
+	ReplicaCount Nullable[int] `json:"replicaCount"`
+}
+
+func TestSimpleInt(t *testing.T) {
+	type testCase struct {
+		name      string
+		jsonInput []byte
+		wantNull  bool
+		wantSet   bool
+	}
+	tests := []testCase{
+		{
+			name:      "simple object: set name to some non null value",
+			jsonInput: []byte(`{"replicaCount":1}`),
+			wantNull:  false,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: set name to empty value",
+			jsonInput: []byte(`{"replicaCount":0}`),
+			wantNull:  false,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: set name to null value",
+			jsonInput: []byte(`{"replicaCount":null}`),
+			wantNull:  true,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: do not provide name in json data",
+			jsonInput: []byte(`{}`),
+			wantNull:  false,
+			wantSet:   false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t1 *testing.T) {
+			var obj SimpleInt
+			err := json.Unmarshal(tt.jsonInput, &obj)
+			assert.NoError(t, err)
+			assert.Equalf(t, tt.wantNull, obj.ReplicaCount.IsNull(), "IsNull()")
+			assert.Equalf(t, tt.wantSet, obj.ReplicaCount.IsSet(), "IsSet()")
+		})
+	}
+}
+
+type SimplePointerInt struct {
+	// cannot decide if it was provided with `null` value in json
+	ReplicaCount Nullable[*int] `json:"replicaCount"`
+}
+
+func TestSimplePointerInt(t *testing.T) {
+	type testCase struct {
+		name      string
+		jsonInput []byte
+		wantNull  bool
+		wantSet   bool
+	}
+	tests := []testCase{
+		{
+			name:      "simple object: set name to some non null value",
+			jsonInput: []byte(`{"replicaCount":1}`),
+			wantNull:  false,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: set name to empty value",
+			jsonInput: []byte(`{"replicaCount":0}`),
+			wantNull:  false,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: set name to null value",
+			jsonInput: []byte(`{"replicaCount":null}`),
+			wantNull:  true,
+			wantSet:   true,
+		},
+
+		{
+			name:      "simple object: do not provide name in json data",
+			jsonInput: []byte(`{}`),
+			wantNull:  false,
+			wantSet:   false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t1 *testing.T) {
+			var obj SimplePointerInt
+			err := json.Unmarshal(tt.jsonInput, &obj)
+			assert.NoError(t, err)
```

Comment:
May be worth replacing this and similar ones with:
```suggestion
			require.NoError(t, err)
```

So we're failing earlier